crates/rpc-types/src/eth/block.rs

```diff
@@ -101,19 +105,32 @@ pub struct Header {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub mix_hash: Option<B256>,
     /// Nonce
-    pub nonce: Option<B64>,
+    #[serde(default, with = "alloy_serde::num::u64_hex_opt")]
+    pub nonce: Option<u64>,
```

Comment:
`skip_serializing_if`

Reply:
I think even nonce is required

Reply:
mix_hash and nonce are options because they are not supported by newer chains that otherwise are compatible with this object.
for convenience I'd like to keep them optional